s3_tools/objects/list.py

```diff
@@ -81,4 +87,7 @@ def list_objects(
 
         continuation_token = response.get("NextContinuationToken")
 
-    return keys if not search_str else fnmatch.filter(keys, search_str)
+    if search_str:
```

Comment:
```suggestion
    if isinstance(search_str, str):
```
otherwise if True or a number (int or float !=0) will be True and raise an exception